cmd/ctl/pkg/create/certificaterequest.go

```diff
@@ -0,0 +1,273 @@
+/*
+Copyright 2020 The Jetstack cert-manager contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package create
+
+import (
+	"context"
+	"encoding/pem"
+	"fmt"
+
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/cli-runtime/pkg/resource"
+	restclient "k8s.io/client-go/rest"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/kubectl/pkg/util/i18n"
+	"k8s.io/kubectl/pkg/util/templates"
+
+	cmapiv1alpha2 "github.com/jetstack/cert-manager/pkg/apis/certmanager/v1alpha2"
+	cmclient "github.com/jetstack/cert-manager/pkg/client/clientset/versioned"
+	"github.com/jetstack/cert-manager/pkg/util/pki"
+	"github.com/jetstack/cert-manager/pkg/webhook"
+)
+
+var (
+	long = templates.LongDesc(i18n.T(`
+Create a cert-manager CertificateRequest resource for one-time Certificate issuing without auto renewal.`))
+
+	example = templates.Examples(i18n.T(`
+# Create a certificate request from from file.
+kubectl cert-manager create certificaterequest -f my-certificate.yaml
+
+# Create a certificate request in namespace sandbox, provided no conflict with namesapce defined in file.
+kubectl cert-manager create certificaterequest --namespace sandbox -f my-certificate.yaml
+
+# Create a certificate request with the name 'my-cr'.
+kubectl cert-manager create certificaterequest -f my-certificate.yaml my-cr
```

Comment:
I would do `kubectl cert-manager create certificaterequest my-cr -f my-certificate.yaml` analog to `create stcert name --from-...`